nixos/modules/services/web-servers/authelia.nix

```diff
@@ -0,0 +1,181 @@
+{ lib
+, pkgs
+, config
+, ...
+}:
+
+let
+  cfg = config.services.authelia;
+
+  format = pkgs.formats.yaml { };
+  configFile = format.generate "config.yml" cfg.settings;
+
+  autheliaOpts = with lib; { name, ... }: {
+    options = {
+      enable = mkEnableOption "Authelia instance";
+
+      name = mkOption {
+        type = types.str;
+        default = name;
+      };
+
+      package = mkOption {
+        default = pkgs.authelia;
+        type = types.package;
+        defaultText = literalExpression "pkgs.authelia";
+      };
+
+      user = mkOption {
+        default = "authelia-${name}";
+        type = types.str;
+      };
+      group = mkOption {
+        default = "authelia-${name}";
+        type = types.str;
+      };
+
+      jwtSecretFile = mkOption {
```

Comment:
I think a generic `environment` option would be better than defining all possible secret files this way.

Reply:
I think a general environment option is good but I think a secrets section is still useful to promote setting them properly.
I'll maybe put these under `secrets.X` and have a separate environment and then merge the results

Reply:
It's also nice to know when you missed setting some, because enabling one might require another.

Reply:
added environmentVariables. moved the secrets under `secrets.X`. Have the sanity check for the 2 required secrets but have an opt-out under `secrets.manual = true;`